src/nbl/video/IOpenGL_PhysicalDeviceBase.h

```diff
@@ -667,18 +750,61 @@ class IOpenGL_PhysicalDeviceBase : public IOpenGLPhysicalDeviceBase
 
 			m_properties.limits.bufferViewAlignment = m_glfeatures.reqTBOAlignment;
 			m_properties.limits.UBOAlignment = m_glfeatures.reqUBOAlignment;
 			m_properties.limits.SSBOAlignment = m_glfeatures.reqSSBOAlignment;
 			
+			GetIntegerv(GL_MIN_PROGRAM_TEXEL_OFFSET, reinterpret_cast<GLint*>(&m_properties.limits.minTexelOffset));
+			GetIntegerv(GL_MAX_PROGRAM_TEXEL_OFFSET, reinterpret_cast<GLint*>(&m_properties.limits.maxTexelOffset));
+			GetIntegerv(GL_MIN_PROGRAM_TEXTURE_GATHER_OFFSET, reinterpret_cast<GLint*>(&m_properties.limits.minTexelGatherOffset));
+			GetIntegerv(GL_MAX_PROGRAM_TEXTURE_GATHER_OFFSET, reinterpret_cast<GLint*>(&m_properties.limits.maxTexelGatherOffset));
+	
+			GetIntegerv(GL_MAX_FRAMEBUFFER_WIDTH, reinterpret_cast<GLint*>(&m_properties.limits.maxFramebufferWidth));
+			GetIntegerv(GL_MAX_FRAMEBUFFER_HEIGHT, reinterpret_cast<GLint*>(&m_properties.limits.maxFramebufferHeight));
+
+			if(!IsGLES || m_glfeatures.Version>=320u)
+				GetIntegerv(GL_MAX_FRAMEBUFFER_LAYERS, reinterpret_cast<GLint*>(&m_properties.limits.maxFramebufferLayers));
+			else if (m_glfeatures.isFeatureAvailable(COpenGLFeatureMap::NBL_OES_geometry_shader))
+				GetIntegerv(GL_MAX_FRAMEBUFFER_LAYERS_OES, reinterpret_cast<GLint*>(&m_properties.limits.maxFramebufferLayers));
+			else if (m_glfeatures.isFeatureAvailable(COpenGLFeatureMap::NBL_EXT_geometry_shader))
+				GetIntegerv(GL_MAX_FRAMEBUFFER_LAYERS_EXT, reinterpret_cast<GLint*>(&m_properties.limits.maxFramebufferLayers));
+
+			GLint maxFramebufferSamples = 0;
+			GetIntegerv(GL_MAX_FRAMEBUFFER_SAMPLES, &maxFramebufferSamples);
+			auto sampleCountFlags = core::bitflag<asset::IImage::E_SAMPLE_COUNT_FLAGS>((0x1u<<(1+core::findMSB(static_cast<uint32_t>(maxFramebufferSamples))))-1u);
+			m_properties.limits.framebufferColorSampleCounts = sampleCountFlags;
+			m_properties.limits.framebufferDepthSampleCounts = sampleCountFlags;
+			m_properties.limits.framebufferStencilSampleCounts = sampleCountFlags;
+			m_properties.limits.framebufferNoAttachmentsSampleCounts = sampleCountFlags;
+
+			m_properties.limits.maxColorAttachments = m_glfeatures.MaxColorAttachments;
 			m_properties.limits.timestampPeriodInNanoSeconds = 1.0f;
 
 			GetFloatv(GL_POINT_SIZE_RANGE, m_properties.limits.pointSizeRange);
 			GetFloatv(GL_ALIASED_LINE_WIDTH_RANGE, m_properties.limits.lineWidthRange);
 			
+			GetFloatv(GL_POINT_SIZE_GRANULARITY, &m_properties.limits.pointSizeGranularity);
+			GetFloatv(GL_LINE_WIDTH_GRANULARITY, &m_properties.limits.lineWidthGranularity);
+
 			m_properties.limits.nonCoherentAtomSize = 256ull;
 			
+
+			/* Vulkan 1.1 Core  */
+			
+			m_properties.limits.maxPerSetDescriptors = m_glfeatures.maxUBOBindings + m_glfeatures.maxSSBOBindings + m_glfeatures.maxTextureBindings + m_glfeatures.maxImageBindings;
+			m_properties.limits.maxMemoryAllocationSize = m_glfeatures.maxBufferSize; // TODO(Erfan): 
```

Comment:
all good, you could maaaybe figure out the largest 2D array and 3D texture bytesizes, but IMHO that's rather pointless